where-filter.ts

```diff
@@ -206,14 +209,21 @@ function test(example, value) {
 	);
 }
 
-function whereFilter(where) {
-	if (typeof where === 'function') {
-		return where;
+export type FilterFunction<T> = (obj: T) => boolean;
+
+// TODO: real types
+// @ts-ignore
+export type Where<T> = any;
+
+export function whereFilter<T>(filter: Where<T> | FilterFunction<T>): FilterFunction<T> {
+	if (typeof filter === 'function') {
+		return filter;
```

Comment:
What's the purpose of supporting passing the filter func through as-is here? That seems like it'd be a mis-use of the library?

Reply:
not sure, "historical reasons" I guess :shrug: 

my strategy was to not change functionality at all for this pass... but none of the usages in warehouse pass a function and this is already going to be marked as a breaking change, so maybe we can just get rid of this weirdness